src/Rector/Core/Utility/MoveApplicationContextToEnvironmentApiRector.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\Core\Utility;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\StaticCall;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Core\Environment;
+
+/**
+ * @see https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.2/Deprecation-89631-UseEnvironmentAPIToFetchApplicationContext.html
+ */
+final class MoveApplicationContextToEnvironmentApiRector extends AbstractRector
+{
+    /**
+     * @inheritDoc
+     */
+    public function getNodeTypes(): array
+    {
+        return [StaticCall::class];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $classNode = $node->class;
+        $className = $this->getName($classNode);
+        $methodName = $this->getName($node);
+
+        if (GeneralUtility::class !== $className) {
+            return null;
+        }
+
+        if ($methodName === 'getApplicationContext') {
```

Comment:
If would prefer this guard clause if(!$this->isName($node->name, 'getApplicationContext')) { return null }